integration/kubernetes/helm-chart/alluxio/templates/proxy/daemonset.yaml

```diff
@@ -0,0 +1,99 @@
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+{{- if .Values.proxy.enabled }}
+{{- $hostNetwork := .Values.proxy.hostNetwork }}
+{{- $hostPID := .Values.proxy.hostPID }}
+{{- $fullName := include "alluxio.fullname" . }}
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ template "alluxio.fullname" . }}-proxy
+  labels:
+    app: {{ template "alluxio.name" . }}
+    chart: {{ template "alluxio.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    role: alluxio-proxy
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "alluxio.name" . }}
+      chart: {{ template "alluxio.chart" . }}
+      release: {{ .Release.Name }}
+      heritage: {{ .Release.Service }}
+      role: alluxio-proxy
+  template:
+    metadata:
+      labels:
+        app: {{ template "alluxio.name" . }}
+        chart: {{ template "alluxio.chart" . }}
+        release: {{ .Release.Name }}
+        heritage: {{ .Release.Service }}
+        role: alluxio-proxy
+    spec:
+      hostNetwork: {{ $hostNetwork }}
+      hostPID: {{ $hostPID }}
```

Comment:
Not sure what the necessity for `hostPID` is. The [k8s docs](https://kubernetes.io/docs/concepts/security/pod-security-policy/#host-namespaces) say the following:
> HostPID - Controls whether the pod containers can share the host process ID namespace. Note that when paired with ptrace this can be used to escalate privileges outside of the container (ptrace is forbidden by default).

Reply:
Yea good catch, `hostPID` requires privileges so many users don't have that by default. If that's not required, we should default to not need it in the deployment.

Reply:
Ok, it looks unnecessary, I will fix this.